src/constraints.jl

```diff
@@ -79,137 +89,80 @@ function Base.showerror(
-
-    add_constraint(model::ModelLike, v::VariableIndex, set::S)::ConstraintIndex{VariableIndex,S} where {S}
-    add_constraint(model::ModelLike, vec::Vector{VariableIndex}, set::S)::ConstraintIndex{VectorOfVariables,S} where {S}
-
-Add the constraint ``v \\in \\mathcal{S}`` where ``v`` is the variable (or vector of variables) referenced by `v` and ``\\mathcal{S}`` is defined by `set`.
-
-* An [`UnsupportedConstraint`](@ref) error is thrown if `model` does not support
-  `F`-in-`S` constraints,
-* a [`AddConstraintNotAllowed`](@ref) error is thrown if it supports `F`-in-`S`
-  constraints but it cannot add the constraint(s) in its current state and
-* a [`ScalarFunctionConstantNotZero`](@ref) error may be thrown if
-  `func` is an `AbstractScalarFunction` with nonzero constant and `set`
-  is [`EqualTo`](@ref), [`GreaterThan`](@ref), [`LessThan`](@ref) or
-  [`Interval`](@ref).
-* a [`LowerBoundAlreadySet`](@ref) error is thrown if `F` is a
-  [`VariableIndex`](@ref) and a constraint was already added to this variable
-  that sets a lower bound.
-* a [`UpperBoundAlreadySet`](@ref) error is thrown if `F` is a
-  [`VariableIndex`](@ref) and a constraint was already added to this variable
-  that sets an upper bound.
+    add_constraint(
+        model::ModelLike,
+        func::F,
+        set::S,
+    )::ConstraintIndex{F,S} where {F,S}
+
+Add the constraint ``f(x) \\in \\mathcal{S}`` where ``f`` is defined by `func`,
+and ``\\mathcal{S}`` is defined by `set`.
+
+## Errors
+
+ * An [`UnsupportedConstraint`](@ref) error is thrown if `model` does not
+   support `F`-in-`S` constraints
+ * An [`AddConstraintNotAllowed`](@ref) error is thrown if `model` supports
+   `F`-in-`S` constraints but it cannot add the constraint in the current state
+ * A [`ScalarFunctionConstantNotZero`](@ref) error may be thrown if `func` is an
+   `AbstractScalarFunction` with nonzero constant and `set` is [`EqualTo`](@ref),
+   [`GreaterThan`](@ref), [`LessThan`](@ref), or [`Interval`](@ref)
+ * A [`LowerBoundAlreadySet`](@ref) error is thrown if `F` is a [`VariableIndex`](@ref)
+   and a constraint was already added to this variable that sets a lower bound
+ * An [`UpperBoundAlreadySet`](@ref) error is thrown if `F` is a [`VariableIndex`](@ref)
+   and a constraint was already added to this variable that sets an upper bound.
 """
 function add_constraint(
     model::ModelLike,
-    func::AbstractFunction,
-    set::AbstractSet,
-)
-    return throw_add_constraint_error_fallback(model, func, set)
-end
-
-# throw_add_constraint_error_fallback checks whether func and set are both
-# scalar or both vector. If it is the case, it calls
-# `correct_throw_add_constraint_error_fallback`
-function throw_add_constraint_error_fallback(
-    model::ModelLike,
-    func::AbstractScalarFunction,
-    set::AbstractScalarSet;
-    kwargs...,
-)
-    return correct_throw_add_constraint_error_fallback(
```

Comment:
And then `correct_throw_add_constraint_error_fallback`, this time with unused `kwargs`